ironfish/src/network/utils/serializers.ts

```diff
@@ -19,26 +20,26 @@ const BLOCK_TRANSACTIONS_LENGTH_BYTES = 2
 
 export function writeBlockHeader(
   bw: bufio.StaticWriter | bufio.BufferWriter,
-  header: SerializedBlockHeader,
+  header: BlockHeader,
 ): bufio.StaticWriter | bufio.BufferWriter {
   bw.writeU32(header.sequence)
   bw.writeHash(header.previousBlockHash)
   bw.writeHash(header.noteCommitment.commitment)
   bw.writeU32(header.noteCommitment.size)
   bw.writeHash(header.nullifierCommitment.commitment)
   bw.writeU32(header.nullifierCommitment.size)
-  bw.writeBytes(BigIntUtils.toBytesLE(BigInt(header.target), 32))
+  bw.writeBytes(BigIntUtils.toBytesLE(BigInt(header.target.targetValue.toString()), 32))
   bw.writeBytes(BigIntUtils.toBytesLE(BigInt(header.randomness), 8))
-  bw.writeU64(header.timestamp)
+  bw.writeU64(header.timestamp.getTime())
 
   Assert.isTrue(BigInt(header.minersFee) <= 0)
   bw.writeBytes(BigIntUtils.toBytesLE(-BigInt(header.minersFee), 8))
 
-  bw.writeBytes(GraffitiSerdeInstance.deserialize(header.graffiti))
+  bw.writeBytes(header.graffiti)
```

Comment:
This no longer does size-checking on the graffiti buffer -- I don't think this should practically be an issue since it should be checked when it's input, and in the block verifier, but maybe worth adding an Assert here just so the code path stays the same

Reply:
Added an Assert for size-checking on the graffiti buffer